source/ranges.tex

```diff
@@ -2343,28 +2389,28 @@
 \item \tcode{\placeholdernc{decay-copy}(E)} if the decayed type of \tcode{E}
 models \libconcept{View}.
 
-\item Otherwise, \tcode{\placeholder{ref-view}\{E\}} if that
-expression is well-formed, where \tcode{\placeholder{ref-view}}
-is the exposition-only \libconcept{View} specified below.
+\item Otherwise, \tcode{ref-view\{E\}} if that expression is well-formed.
 
 \item Otherwise, \tcode{subrange\{E\}}.
 \end{itemize}
 
-\rSec3[range.view.ref]{\placeholder{ref-view}}
+\rSec3[range.ref.view]{Class template \tcode{ref_view}}
 
+\pnum
+\tcode{ref_view} is a \tcode{View} of the elements of some other \tcode{Range}.
 \begin{codeblock}
 namespace std::ranges {
   template<Range R>
     requires is_object_v<R>
-  class @\placeholder{ref-view}@ : public view_interface<@\placeholder{ref-view}@<R>> {
+  class ref_view : public view_interface<ref_view<R>> {
```

Comment:
The ingress paper makes this previously exposition-only class template a publicly-exported class template, but fails to add it to any header synopsis. I suspect the author's intent is that it be declared in [ranges.syn]:
```diff
 // \ref{range.all}, all view
 namespace view { inline constexpr @\unspec@ all = @\unspec@; }
 
 template<ViewableRange R>
   using all_view = decltype(view::all(declval<R>()));
 
+template<Range R>
+  requires is_object_v<R>
+class ref_view;
 
 // \ref{range.filter}, filter view
```

Reply:
@CaseyCarter and @johelegp, how should we resolve the ref_view issue?  I'm marking this "changes required" in the mean time so that we won't merge it as is.  Might want to send an email to library chair, as we only have a few more days before the post mailing.

Reply:
Everything that's left seems pretty much editorial to me.

Reply:
I'd be fine with simply inserting the declaration in the synopsis (disclaimer: I am the author of this paper), but if you'd like approval from the LWG chair I can poke Marshall.

Reply:
Fixed.